packages/gatsby-source-wordpress/src/steps/create-schema-customization/build-types.js

```diff
@@ -29,12 +30,14 @@ const unionType = typeBuilderApi => {
     name: buildTypeName(type.name),
     types,
     resolveType: node => {
-      if (node.type) {
-        return buildTypeName(node.type)
-      }
-
       if (node.__typename) {
-        return buildTypeName(node.__typename)
+        const {
+          schema: { typePrefix: prefix },
+        } = getPluginOptions()
+
+        return node.__typename.startsWith(prefix)
+          ? node.__typename
+          : buildTypeName(node.__typename)
```

Comment:
Doesn't `buildTypeName` already do this logic for you?

Reply:
Yup indeed, thanks for noticing :)